source/contributed/extending-prototypes.md

```diff
@@ -0,0 +1,75 @@
+title: Extending Object Prototypes
+---
+
+Extending prototypes is generally something you should avoid doing in most cases, however Screeps encourages it. This article will take you through the basics of how, when, and why.
+
+Generally the reason you will be adding to the prototype is for convenience. Other times will be because your property or method doesn't fit anywhere else in your code. An example of this is adding properties to the Room prototype to access the room's mineral and sources.
+
+```
+Object.defineProperties(Room.prototype, {
+    mineral: {
+        // The configurable option means you can change the property later
+        // Like on the next tick. It's important we set this to true.
+        configurable: true,
+        get() {
+            if (!_.isUndefined(this._mineral)) {
+                [this._mineral] = this.find(FIND_MINERALS);
+            }
+            return this._mineral;
+        },
+    },
+
+    sources: {
+        configurable: true,
+        get() {
+            if (!_.isUndefined(this._sources)) {
+                this._sources = this.find(FIND_SOURCES);
+            }
+            return this._sources;
+        },
+    }
+});
+```
+
+These properties can now be accessed using `room.mineral` or `room.sources`.
+
+If you want to create a method instead of a property, you have two methods. Use the ES6 `Object.defineProperty` syntax that was used above, or use the classic method. I will show you both below.
+
+```
+// ES6 Object.defineProperty
+Object.defineProperty(Room.prototype, 'calcGlobalCoordinates', {
+    configurable: true,
+    value: function() {
+        const parsed = /^[WE]([0-9]+)[NS]([0-9]+)$/.exec(this.name);
+        const x = +parsed[1];
+        const y = +parsed[2];
+        return {x, y};
+    }
+});
+
+// Classic Prototype Extend
+Room.prototype.calcGlobalCoordinates = function() {
+    const parsed = /^[WE]([0-9]+)[NS]([0-9]+)$/.exec(this.name);
```

Comment:
W and N coordinates should be translated by -1.

Or maybe find a different example ;)